bastion installation on Fedora 31

ostefek99 opened the pull request:
### What does this PR do?
Docker requires cgroups v1 to operate. Fedora 31 uses cgroup v2 by default.
Added tasks to enable kernel parameters for cgroups v1.

### How should this be tested?
Run the automation.

### Is there a relevant Issue open for this?
n/a

### Other Relevant info, PRs, etc.
n/a

### People to notify
cc: @redhat-cop/infra-ansible

oybed commented:
@ostefek99 Can we please make this optional - i.e.: configurable? Don't know that we'd want this to "always" be applied when running Fedora. Also, I see the code is explicitly checking for F31... that's not good for next iterations of Fedora as it will require additional PRs to support. I'd think that if we made this option configurable, we don't need to make that check for the OS (or version), but rather just apply it if set.

oybed commented on roles/config-docker/tasks/main.yml:
```diff
@@ -1,6 +1,11 @@
+  - name: Configure Fedora 31 to use cgroups v1
+    import_tasks: fedora_kernel_cgroups.yml
+    when:
+    - (ansible_distribution == "Fedora") and (ansible_distribution_version == "31")
```
As mentioned in the overall comment, I'd rather see a solution here that checks if the user requested to use `cgroup_v1` rather than if this is running on Fedora (and 31). In that way, the implementation is more generic and allows it to be applied on other OS types and versions as well.

ostefek99 replied:
It was intended to be a quick fix for the Fedora 31 since we have no idea what will change in the future (Docker will adopt cgroups v2 or Fedora 3x will revert back to cgroup v1 - unlikely but you never know). It makes sense to have it for other OSes as well but this requires more research since the task uses grubby tool to make those changes which might not be available at least out of the box. Let me dig in further.

ostefek99 commented:
Changes:
- changed it to be more general
- add new variable in bastion.yml - **kernel_cgroups_v1**
- update README.md to reflect a new variable **kernel_cgroups_v1** (boolean)
- renamed task (in roles/config-docker/tasks/main.yml) and the _fedora_kernel_cgroups.yml_ file to a more general name _kernel_cgroups_v1.yml_

oybed commented on roles/config-docker/tasks/kernel_cgroups_v1.yml:
```diff
@@ -0,0 +1,13 @@
+- name: Update Grub config file
+  shell: grubby --args="{{ grub_add_arg }}" --update-kernel=ALL
+
+- name: Reboot the host
```
This will "always" reboot the host as written, even if the above grubby line didn't change anything (i.e.: the host already uses v1). A good follow-up would be to make the reboot happen only if a change was made. Not sure if `grubby` will provide a returncode that can be used for this, but another way would be to use the ansible lineinfile module instead - that would also help with the fact that `grubby` isn't always installed (which should be another enhancement if we continue to use `grubby` - i.e.: ensure it's installed before using it).

oybed commented on roles/config-docker/tasks/kernel_cgroups_v1.yml:
```diff
@@ -0,0 +1,13 @@
+  poll: 0
+  ignore_errors: true
+
+- name: Waiting for the server to come back
```
Could be good to consider re-using another role we have for reboot/wait on host - i.e.: https://github.com/redhat-cop/infra-ansible/tree/master/roles/update-host.
